setup.cfg

```diff
@@ -34,6 +34,7 @@ install_requires =
     voila>=0.3.5
     pyyaml>=5.4.1
     specutils>=1.7.0
+    specreduce>=1.0.0
```

Comment:
specreduce will be an added dependency (although we might need to pin a newer version).

Reply:
Yet another required dependency... Is @havok2063 okay with this?

Reply:
If not, we could have it be required for the plugin to show up, but then users would need to install additional dependencies depending on which plugins they want.

Reply:
Tricky business: #1399

Reply:
I think I'm ok with it, since we will use this plugin in MAST.  Does this plugin also provide users information on the default spectral extraction that occurs when I load a single s2d file?  Or is it only for performing custom extractions?

Reply:
As far as I know, the (current) plan is only for custom extraction and using jdaviz as a visualization layer to existing and upcoming functionality in specreduce.

Reply:
User feedback said they wanted to know how the default extraction was made.  If we switch the default extraction to use this plugin to perform the extraction on initial load, would that provide that info and satisfy this feedback?

Reply:
That's an interesting idea.  I think the current displayed spectrum is just collapsed across the entire image.  But if we could come up with automated defaults as input into the specreduce functionality, use those for the initial collapse instead, and have those set as the default in the plugin, that would be slick!  I'm guessing even if that extraction isn't perfect, it would be better than just the full collapse?

There already is a snackbar warning that says: "The collapsed 1D spectrum is for quicklook purposes only. A robust extraction should be used for scientific use cases." Perhaps this text should also be updated to direct the user to this new plugin?  Or maybe the "Spectrum 1D" data label should be changed to "Collapsed Spectrum"?

Reply:
Yeah I think it would be fine to define some reasonable defaults to perform a default extraction.  We would still display the snackbar warning message.  Yeah it's also a good idea to update the message to point people to the plugin.  If it's too much work to switch over, then I am also ok just updating the label to say "Collapsed Spectrum" and updating the message to point people to the plugin.